internal/controller/config/config_test.go

```diff
@@ -0,0 +1,53 @@
+package config_test
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/argoproj-labs/ephemeral-access/internal/controller/config"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConfiguration(t *testing.T) {
+	t.Run("will validate if default values are set properly", func(t *testing.T) {
+		// When
+		config, err := config.ReadEnvConfigs()
+
+		// Then
+		assert.NoError(t, err, "NewConfiguration error")
+		assert.Equal(t, "info", config.LogLevel())
+		assert.Equal(t, "console", config.LogFormat())
+		assert.Equal(t, "0", config.MetricsAddress())
+		assert.Equal(t, false, config.MetricsSecure())
+		assert.Equal(t, false, config.EnableLeaderElection())
+		assert.Equal(t, ":8081", config.ControllerHealthProbeAddr())
+		assert.Equal(t, false, config.ControllerEnableHTTP2())
+		assert.Equal(t, time.Minute*3, config.ControllerRequeueInterval())
+	})
+	t.Run("will validate if env vars are set properly", func(t *testing.T) {
+		// Given
+		os.Setenv("EPHEMERAL_LOG_LEVEL", "debug")
```

Comment:
You should use `t.Setenv` instead of `os.Setenv`. It will automatically perform env cleanup between tests and fail if they run in parallel.